### PR TITLE
MAINT: Apply ruff/pycodestyle warning rules (W)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
In preparation for https://github.com/numpy/numpy/issues/24994.

These commits are very basic, but modify many files. Nevertheless, I think these `W` rules would be a nice addition.

### CODE REVIEW

File: doc/source/user/plots/matplotlib1.py [hunk 1]
```diff
@@ -3,5 +3,5 @@
 
 a = np.array([2, 1, 5, 7, 4, 6, 8, 14, 10, 9, 18, 20, 22])
 
-plt.plot(a) 
-plt.show()
\ No newline at end of file
+plt.plot(a)
+plt.show()
```
Comment (charris) [inline doc/source/user/plots/matplotlib1.py:7]: Are these removing newlines? The Unix standard is that text files end in newlines.

**Comment by charris** [comment]
Mostly LGTM. I don't mind removing trailing white space, I do it occasionally myself.

File: doc/source/user/plots/matplotlib1.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply doc/source/user/plots/matplotlib1.py:7]: No, they add the missing newline at EOF.

**Comment by DimitriPapadopoulos** [comment]
Note that I am merely running `pycodestyle` on the existing codebase, to align with newer commits that are linted with `pycodestyle`.

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
Thanks @DimitriPapadopoulos .

**Pull request merged**